### ISSUE TITLE
[Feature Request][WEBJS] Un/Archived webhook event type

### ISSUE THREAD

**Issue by arielpts** [allburov/whatsapp-http-api#434]
**Is your feature request related to a problem? Please describe.** I'm always
> We sync "active" conversations between the CRM and WAHA. Whatsapp has the "archived: bool" flag that can be use precisely to set a conversation as active or inactive, and the app can be configured to automatically unarchive conversations on new messages, like other contact centers software.

**Describe the solution you'd like**
> An ideal solution would be to create a new event type such as `chat.archived` to be sent via webhook when the chat is archived/unarchived.

**Describe alternatives you've considered**
> In order to sync the the chats that are archived or not, currently we need to pool the endpoint `/chats` and read the property `"archived": true/false`. Polling this resource is a slow and resource intensive process.

**Additional context**
> whatsapp-web.js already has a `chat_archived` event that is emitted when a chat is archived/unarchived, so the implementation of this webhook hopefully should not be difficult.
> - https://docs.wwebjs.dev/Client.html#event:chat_archived
> - https://github.com/devlikeapro/whatsapp-web.js/blob/41b280c769918db03ac6ec12cf0e4215e7b864ef/index.d.ts#L344

_PS: We stopped using WAHA early this year because we migrated to the official API and now we resumed because of another use case and, honestly, we rather use the WAHA API than the WA one, also the whole product evolved a lot in the last 6 months, kudos to the team._

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment]
@arielpts you're using WEBJS engine right? Just wanna make sure to prioritize it better

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment]
Thank you for the kind words and your support, it means the world to us :heart_hands: :heart_eyes_cat: 

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by arielpts** [comment]
Yes, we always use WEBJS 🧡 

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment]
Available in [2024.8.1](https://waha.devlike.pro/blog/waha-2024.8-whatsapp-business-labels-api-chat-archive-and-more/)

----
👉 Read more about the endpoints on [💬 Chats](https://waha.devlike.pro/docs/how-to/chats/#archive-chat) page!

### Archive chat

Use the method to archive chat
```bash
POST /api/{session}/chats/{chatId}/archive
```

### Unarchive chat

Use the method to unarchive chat

```bash
POST /api/{session}/chats/{chatId}/unarchive
```

### chat.archive
The `chat.archive` event is triggered when a chat is archived or unarchived.
```json
{
  "event": "chat.archive",
  "session": "default",
  "payload": {
    "id": "123123123@c.us",
    "timestamp": 1667561485,
    "archived": true <== or false
  },
  ...
}
```


[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)
